mdx/guides/0x-api-faq.mdx

```diff
@@ -0,0 +1,142 @@
+
+It is possible for the protocol fees to be paid in WETH rather than sent in the transaction as ETH. This can be achieved by the taker having a WETH balance and setting a WETH allowance to the Protocol fee collector address. This [address](https://etherscan.io/address/0xa26e80e7dea86279c6d778d702cc413e6cffa777) can be read on the Exchange contract: `Exchange.protocolFeeCollector`.
+
+**TLDR** 0x API handles the heavy lifting related to protocol fees and provides a `value` field.
+
+## Why does my 0x transaction revert?
+
+Aw shucks, are 0x quotes reverting? There are a number of low hanging fruits to check for when attempting to execute a quote beyond the obvious revert issues related to eth transactions:
+
+* Are allowances properly set for the user to trade the `sellToken`?
+* Does the user have enough `sellToken` balance to execute the swap?
+* Do users have enough ETH or WETH to pay the `protocolFee`?
+
+## How does `takerAddress` help with catching issues?
+
+By passing a `takerAddress` parameter, 0x API can provide a more bespoke quote and help catch revert issues:
+
+* 0x API will estimate the gas cost for `takerAddress` to execute the provided quote.
+* If successfully called, the `gas` parameter in the quote will be an accurate amount of gas needed to execute the swap.
+* If unsuccessful for revert reasons suggested above, then 0x API will throw a gas cost estimation error, alluding to an issue with the `takerAddress` executing the quote.
+
+**TLDR** Pass `takerAddress` to get the quote validated before provided to you, assuring that a number of revert cases will not occur.
+
+## 0x orders are reverting but my transaction is fine, what is happening?
+
+Developers may note when analyzing their transactions that some subset of 0x orders may revert (not filled) but the whole transaction is successful. This is expected behavior as implied earlier, some orders due to timing, and the pricing may be filled or expired before a users attempt to fill the order. This would result in a revert and 0x protocol will utilize fallback orders to compensate for the reverted order. This will result in a successful transaction even though reverts occur within the transactions.
+
+## How does gas price influence my transaction?
+
+It is important to set a higher gas price as a default rather than to set it low and rely on the user to increase this in their wallet. The gas price directly influences the 0x Protocol fee paid to market makers. The 0xAPI calculates this protocol fee for you (and returns it in the `value` field) based on the supplied `gasPrice` (defaulted to `fast`). But if a gas price of 1 gwei was supplied and the user changes this to any higher amount, the transaction will fail as the `value` has not been updated to reflect the new gas price for the protocol fee.
```

Comment:
```suggestion
It is important to set a higher gas price as a default rather than to set it low and rely on the user to increase this in their wallet. The gas price directly influences the 0x Protocol fee paid to market makers. The 0x API calculates this protocol fee for you (and returns it in the `value` field) based on the supplied `gasPrice` (defaulted to `fast`). But if a gas price of 1 gwei was supplied and the user changes this to any higher amount, the transaction will fail as the `value` has not been updated to reflect the new gas price for the protocol fee.
```